pkg/ottl/ottlfuncs/func_remove_xml.go

```diff
@@ -0,0 +1,94 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/antchfx/xmlquery"
+	"github.com/antchfx/xpath"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type RemoveXMLArguments[K any] struct {
+	Target ottl.StringGetter[K]
+	XPath  string
+}
+
+func NewRemoveXMLFactory[K any]() ottl.Factory[K] {
+	return ottl.NewFactory("RemoveXML", &RemoveXMLArguments[K]{}, createRemoveXMLFunction[K])
+}
+
+func createRemoveXMLFunction[K any](_ ottl.FunctionContext, oArgs ottl.Arguments) (ottl.ExprFunc[K], error) {
+	args, ok := oArgs.(*RemoveXMLArguments[K])
+
+	if !ok {
+		return nil, fmt.Errorf("RemoveXML args must be of type *RemoveXMLAguments[K]")
+	}
+
+	return removeXML(args.Target, args.XPath), nil
+}
+
+// removeXML returns a `pcommon.String` that is a result of removing all matching nodes from the target XML.
+// This currently supports removal of elements, attributes, text values, comments, and CharData.
+func removeXML[K any](target ottl.StringGetter[K], xPath string) ottl.ExprFunc[K] {
+	return func(ctx context.Context, tCtx K) (any, error) {
+		if err := validateXPath(xPath); err != nil {
+			return nil, err
+		}
```

Comment:
since `xPath` is static we can validate it once during startup by moving this check before the return statement.